Jenkinsfile

```diff
@@ -377,15 +411,22 @@ stage('Default build') {
 	}
 	runBuildOnNode( NODE_PATTERN_BASE, [time: 2, unit: 'HOURS'] ) {
 		withMavenWorkspace(mavenSettingsConfig: deploySnapshot ? helper.configuration.file.deployment.maven.settingsId : null) {
+			if ( incrementalBuild ) {
+				dir(helper.configuration.maven.localRepositoryPath) {
+					unstash name:'default-build-result'
+				}
+			}
+			// If we are in the pull request we've already run JQAssistant and source formatting checks
 			String mavenArgs = """ \
 					--fail-at-end \
-					-Pdist -Pcoverage -Pjqassistant -Pci-sources-check \
+					-Pdist -Pcoverage \
+					${incrementalBuild ? ('-Dincremental -Dgib.referenceBranch=refs/remotes/origin/'+helper.scmSource.pullRequest.target.name) : '-Pjqassistant -Pci-sources-check'} \
```

Comment:
Oh, but that would require running unit tests in the first stage and not here, and then they wouldn't be included in Sonar test coverage reports... That would be annoying.

I wonder what happens if we just execute Sonar in every single stage... Would Sonar aggregate the results? :thinking:
If so that would be really great as we would also be able to execute sonar in (BEFORE_MERGE) non-default environements and would get test coverage numbers that more closely reflect reality.

That being said, I also wonder if Sonar test coverage numbers will still be relevant with incremental builds... Let's hope that Sonar only considers test coverage on changed code.

Reply:
Okay that's complex... it may be a separate problem to handle in a separate issue, e.g. "Build public modules only once in PR builds on CI" and maybe also "Enable Sonar test coverage on non-default environment builds on CI". WDYT, should I open those?